src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -992,6 +992,61 @@ class VariantSampleMatrix[T](val metadata: VariantMetadata,
         "sa" -> saSignature),
       Array("s"))
   }
+
+  def annotateVariantsKeyTable(kt: KeyTable, code: String) = {
+    val ktKeyTypes = kt.keySignature.fields.map(_.`type`)
+
+    if (ktKeyTypes.size != 1 || ktKeyTypes(0) != TVariant)
+      fatal(s"Key signature of KeyTable must be 1 field with type `Variant'. Found `${kt.keySignature}'")
+
+    val ktSig = kt.signature
+
+    val inserterEc = EvalContext(Map("va" -> (0, vaSignature), "table" -> (1, ktSig)))
+
+    val (finalType, inserter) =
+      buildInserter(code, vaSignature, inserterEc, Annotation.VARIANT_HEAD)
+
+    val keyedRDD = kt.rdd.map { case (k: Row, v) => (k(0).asInstanceOf[Variant], kt.mergeKeyAndValue(k, v)) }
+
+    val ordRdd = OrderedRDD(keyedRDD, None, None)
+
+    annotateVariants(ordRdd, finalType, inserter)
+  }
+
+  def annotateVariantsKeyTable(kt: KeyTable, vdsKey: String, code: String) = {
+
+    val vdsKeyEc = EvalContext(Map("v" -> (0, TVariant), "va" -> (1, vaSignature)))
+
+    val (vdsKeyType, vdsKeyF) = Parser.parseExpr(vdsKey, vdsKeyEc)
+
+    if (kt.keySignature != vdsKeyType)
+      fatal(s"Key signature of KeyTable, `${kt.keySignature}', must match type of computed key, `${vdsKeyType}'.")
+
+    val ktSig = kt.signature
+
+    val inserterEc = EvalContext(Map("va" -> (0, vaSignature), "table" -> (1, ktSig)))
+
+    val (finalType, inserter) =
+      buildInserter(code, vaSignature, inserterEc, Annotation.VARIANT_HEAD)
+
+    val ktRdd = kt.rdd.map { case (k, v) => (k, kt.mergeKeyAndValue(k, v)) }
```

Comment:
This seems a reasonable approach to me. I was a little confused by the interface at first, but I don't have any strong/clear opinions on how to reform it. I'm happy to fix this up when we have a new interface.